src/wellKnownDidConfiguration/index.ts

```diff
@@ -70,16 +76,16 @@ export async function createCredential(
 
   const credential = Credential.fromClaim(claim)
 
-  const assertionKey = document.assertionMethod?.[0]
-
-  if (!assertionKey) {
-    throw new Error('Full DID doesnt have assertion key: Please add assertion key')
-  }
-
-  return Credential.createPresentation({
+  const presentation = await Credential.createPresentation({
     credential,
     signCallback,
   })
+
+  if (presentation.claimerSignature.keyUri !== document.uri + assertionKey.id) {
+    throw new Error('The credential presentation needs to be signed with the assertionMethod key')
```

Comment:
How can we even get into this situation?

Reply:
the sign callback is to be implemented by the caller of this function, and they could mess up

Reply:
Oh, right, I forgot that this can be called as library code, not via CLI tool